realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -404,6 +404,192 @@ public boolean isValid() {
         return this;
     }
 
+    // In
+
+    /**
+     * In comparison. This allows you to test if objects match any value in an array of values.
+     *
+     * @param fieldName the field to compare.
+     * @param values values to compare with and it cannot be null or empty.
+     * @return the query object.
+     * @throws java.lang.IllegalArgumentException if values is empty or do not match class or field type.
+     */
+    public RealmQuery<E> in(String fieldName, String... values) {
```

Comment:
I know that the var arg variant can also be written with arrays, but a better solution for that would actually be `in(String field Name, Case case, String first field, String... additionalFields)` (see Effective Java item 42), since that would enforce at least one argument.

However moving the Case parameter feels awkward to me, especially since I don't think we should promote the non-array version. Just because something is possible doesn't mean we have to do it, especially if readability and consistency suffers 😄 

But would be nice to hear from @zaki50 why the var arg version is better.